packages/react-router-dom/index.tsx

```diff
@@ -1318,11 +1318,15 @@ function useScrollRestoration({
         let key = (getKey ? getKey(location, matches) : null) || location.key;
         savedScrollPositions[key] = window.scrollY;
       }
-      sessionStorage.setItem(
-        storageKey || SCROLL_RESTORATION_STORAGE_KEY,
-        JSON.stringify(savedScrollPositions)
-      );
-      window.history.scrollRestoration = "auto";
+      try {
+        sessionStorage.setItem(
+          storageKey || SCROLL_RESTORATION_STORAGE_KEY,
+          JSON.stringify(savedScrollPositions)
+        );
+        window.history.scrollRestoration = "auto";
+      } catch (error) {
+        console.warn("Failed to record scroll position in session storage. Scroll restoration will not work.", error);
+      }
```

Comment:
why call `console.error`? I guess it's to make sure the cause is displayed somewhere, but since we can recover from the situation I don't think it should be reported as an error - could it remain `console.warn`?

Reply:
Yeah the idea there was to show the actual underlying error, but we could probably inline `error.message` into the warning as well if we wanted to keep just one console entry and avoid a `console.error`.